searx/engines/lemmy.py

```diff
@@ -0,0 +1,203 @@
+                'metadata': metadata,
+            }
+        )
+    return results
+
+
+def _get_users(json):
+    results = []
+
+    for result in json["users"]:
+        results.append(
+            {
+                'url': result['person']['actor_id'],
+                'title': result['person']['name'],
+                'content': _format_content(result['person'].get('bio', '')),
+            }
+        )
+
+    return results
+
+
+def _get_posts(json):
+    results = []
+
+    for result in json["posts"]:
+        user = result['creator'].get('display_name', result['creator']['name'])
+
+        img_src = None
+        if result['post'].get('thumbnail_url'):
+            img_src = result['post']['thumbnail_url'] + '?format=webp&thumbnail=128'
+
+        metadata = (
+            f"&#x25B2; {result['counts']['upvotes']} &#x25BC; {result['counts']['downvotes']}"
+            f" | {gettext('user')}: {user}"
+            f" | {gettext('comments')}: {result['counts']['comments']}"
+            f" | {gettext('community')}: {result['community']['title']}"
+        )
+
+        content = result['post'].get('body', '').strip()
+        if content:
+            content = _format_content(content)
+
+        results.append(
+            {
+                'url': result['post']['ap_id'],
+                'title': result['post']['name'],
+                'content': content,
+                'img_src': img_src,
+                'publishedDate': datetime.strptime(result['post']['published'][:19], '%Y-%m-%dT%H:%M:%S'),
+                'metadata': metadata,
+            }
+        )
+
+    return results
+
+
+def _get_comments(json):
+    results = []
+
+    for result in json["comments"]:
+        user = result['creator'].get('display_name', result['creator']['name'])
+
+        content = result['comment'].get('content', '').strip()
+        if content:
+            content = _format_content(content)
+
+        metadata = (
+            f"&#x25B2; {result['counts']['upvotes']} &#x25BC; {result['counts']['downvotes']}"
+            f" | {gettext('user')}: {user}"
+            f" | {gettext('community')}: {result['community']['title']}"
+        )
+
+        results.append(
+            {
+                'url': result['comment']['ap_id'],
+                'title': result['post']['name'],
+                'content': _format_content(result['comment']['content']),
+                'publishedDate': datetime.strptime(result['comment']['published'][:19], '%Y-%m-%dT%H:%M:%S'),
+                'metadata': metadata,
+            }
+        )
+
+    return results
+
+
+def response(resp):
+    json = resp.json()
+
+    if lemmy_type == "Communities":
+        return _get_communities(json)
+
+    if lemmy_type == "Users":
+        return _get_users(json)
+
+    if lemmy_type == "Posts":
+        return _get_posts(json)
+
+    if lemmy_type == "Comments":
+        return _get_comments(json)
+
+    raise ValueError("Unsupported lemmy type!")
```

Comment:
I've seen, we usually do `return []` if the selected filter in `settings.yml` is not supported, do you think it's generally better to raise an error or just ignore it and return nothing?

e.g. https://github.com/searxng/searxng/blob/473129031710973e3c02e7178085b751b5e821ed/searx/engines/google_play.py#L51

Reply:
Since it is a misconfiguration I think it is better to throw an exception .. may we should fix the google engine also.  But we should better include the invalid value in the exception message:

```python
raise ValueError("Unsupported lemmy type: %s", lemmy_type)
```